cmd/argocd/commands/root.go

```diff
@@ -94,3 +97,40 @@ func NewCommand() *cobra.Command {
 
 	return command
 }
+
+// HandleCommandExecutionError processes the error returned from executing the command.
+// It handles both standard Argo CD commands and plugin commands. We don't require to return
+// error but we are doing it to cover various test scenarios.
+func HandleCommandExecutionError(err error, isArgocdCLI bool, o ArgoCDCLIOptions) error {
+	cli.SetLogFormat("text")
+	cli.SetLogLevel("info")
+	if err != nil {
+		// If it's an unknown command error, attempt to handle it as a plugin.
+		// Unfortunately, cobra doesn't handle this error, so we need to assume
+		// that error consists of substring "unknown command".
+		// https://github.com/spf13/cobra/pull/2167
+		if isArgocdCLI && strings.Contains(err.Error(), "unknown command") {
+			// The PluginPath is important to be returned since it
+			// helps us understanding the logic for handling errors.
+			PluginPath, pluginErr := HandlePluginCommand(o.PluginHandler, o.Arguments[1:], 1)
```

Comment:
Once `HandleCommandExecutionError` has been moved to `plugin.go` I think we should do one of two things:

1. Either inline the `HandlePluginCommand` code into this function (since it's not used anywhere else)
2. Alternatively, and perhaps that is more preferable, make `HandlePluginCommand` private (i.e. lowercase the function to `handlePluginCommand`), since this is not intended to be used anywhere else than in `HandleCommandExecutionError`.